test/ProblemDetailsResponseFactoryTest.php

```diff
@@ -176,4 +178,14 @@ public function testFactoryRendersPreviousExceptionsInDebugMode() : void
         $this->assertEquals(101010, $payload['exception']['stack'][0]['code']);
         $this->assertEquals('first', $payload['exception']['stack'][0]['message']);
     }
+
+    public function testFragileDataInExceptionMessageShouldBeHiddenInResponseBodyInNoneDebugMode()
+    {
+        $fragileMessage = 'Your SQL or password here';
+        $exception = new \Exception($fragileMessage);
+
+        $response = $this->factory->createResponseFromThrowable($this->request->reveal(), $exception);
+
+        $this->assertNotContains($fragileMessage, (string) $response->getBody());
```

Comment:
Let's make it a flag, then, and have it default to not use non-`ProblemDetailsException` messages; additionally, there should be a constructor argument for specifying the default message to use with such exceptions.